### PR TITLE
Update the metrics compliance matrix for java.

### PR DESCRIPTION
**Pull request by jkwatson**
## Changes
Add entries for Java in the metrics compliance matrix
(Also includes a couple of items from tracing that were missing)

### CODE REVIEW

**Review by carlosalberto — APPROVED** [review]

File: spec-compliance-matrix.md [hunk 1]
```diff
@@ -96,85 +96,85 @@ Disclaimer: this list of features is still a work in progress, please refer to t
...
+| The meter provides functions to create a new `UpDownCounter`.                                                                                                                |          |    |  +   |    |        |      |        |     |      |     |   -  |       |
+| The meter provides functions to create a new `AsynchronousUpDownCounter`.                                                                                                    |          |    |  +   |    |        |      |        |     |      |     |   -  |       |
+| Instruments have `name`                                                                                                                                                      |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
 | Instruments have kind.                                                                                                                                                       |          |    |      |    |        |      |        |     |      |     |   +  |       |
```
Comment (jsuereth) [inline spec-compliance-matrix.md:113]: This is named pretty poorly, 

It comes from [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#instrument).  

We do have the kind available via the builder we use to construct the instrument, and the kind is available in the SDK.

File: spec-compliance-matrix.md [hunk 2]
```diff
@@ -96,85 +96,85 @@ Disclaimer: this list of features is still a work in progress, please refer to t
...
+| The API for `AsynchronousCounter` accepts the unit of the instrument.                                                                                                        |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The API for `AsynchronousCounter` accepts the description of the instrument.                                                                                                 |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The API for `AsynchronousCounter` accepts a callback.                                                                                                                        |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| `AsynchronousCounter` has an `add` method.                                                                                                                                   |          |    |  -   |    |        |      |        |     |      |     |      |       |
```
Comment (jsuereth) [inline spec-compliance-matrix.md:135]: This isn't in the specification.... where did this come from?

File: spec-compliance-matrix.md [hunk 3]
```diff
@@ -96,85 +96,85 @@ Disclaimer: this list of features is still a work in progress, please refer to t
...
+| The `add` method accepts optional attributes.                                                                                                                                |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The `add` method accepts the increment amount.                                                                                                                               |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The `add` method of `Counter` accepts only positive amounts.                                                                                                                 |          |    |  +   |    |        |      |        |     |      |     |   -  |       |
+| `create_asynchronous_counter` returns an `AsynchronousCounter` object.                                                                                                       |          |    |  -   |    |        |      |        |     |      |     |   +  |       |
```
Comment (jsuereth) [inline spec-compliance-matrix.md:130]: This also isn't in the specification.

File: spec-compliance-matrix.md [hunk 4]
```diff
@@ -96,85 +96,85 @@ Disclaimer: this list of features is still a work in progress, please refer to t
...
-| The metrics exporter `ForceFlush` can inform the caller wether it succeeded, failed or timed out.                                                                            |          |    |      |    |        |      |        |     |      |     |   +  |       |
-| The metrics exporter provides a `shutdown` function.                                                                                                                         |          |    |      |    |        |      |        |     |      |     |   +  |       |
-| The metrics exporter `shutdown` function do not block indefinitely.                                                                                                          |          |    |      |    |        |      |        |     |      |     |   +  |       |
+| The callback function of an `Asynchronous` instrument does not block indefinitely.                                                                                           |          |    |  -   |    |        |      |        |     |      |     |   ?  |       |
```
Comment (jsuereth) [inline spec-compliance-matrix.md:140]: This is a requirement on instrumentation, not the SDK.... we should figure out how to word this appropriately in the SPec.

File: spec-compliance-matrix.md [hunk 5]
```diff
@@ -96,85 +96,85 @@ Disclaimer: this list of features is still a work in progress, please refer to t
...
+| A specified `Resource` can be associated with all the produced metrics from any `Meter` from the `MeterProvider`.                                                            |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The supplied `name`, `version` and `schema_url` arguments passed to the `MeterProvider` are used to create an `InstrumentationLibrary` instance stored in the `Meter`.       |          |    |  +   |    |        |      |        |     |      |     |   -  |       |
+| Configuration can be managed by the `MeterProvider`.                                                                                                                         |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The `MeterProvider` provides methods to update the configuration                                                                                                             |          |    |  -   |    |        |      |        |     |      |     |   +  |       |
```
Comment (jsuereth) [inline spec-compliance-matrix.md:150]: These are optional requirements.  All "MAY" requirements were supposed to be removed from the matrix.   I'd either flip the optional column here, or remove the whole row.

File: spec-compliance-matrix.md [hunk 6]
```diff
@@ -96,85 +96,85 @@ Disclaimer: this list of features is still a work in progress, please refer to t
...
+| The `Histogram` aggregation performs as specified.                                                                                                                           |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The explicit bucket `Histogram` aggregation is available.                                                                                                                    |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The explicit bucket `Histogram` aggregation performs as specified.                                                                                                           |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The metrics exporter has access to the pre aggregated metrics data.                                                                                                          |          |    |  -   |    |        |      |        |     |      |     |   +  |       |
```
Comment (jsuereth) [inline spec-compliance-matrix.md:169]: Curious why this is marked as a `-`?   I think this requirement is hella confusing.  TL;DR; if we send java's `MetricData` to an exporter, we've already aggregated the data.   This was meant to say exporters see data *after* aggregation, not before.

File: spec-compliance-matrix.md [hunk 7]
```diff
@@ -96,85 +96,85 @@ Disclaimer: this list of features is still a work in progress, please refer to t
...
+| The metrics exporter `export` function does not block indefinitely.                                                                                                          |          |    |  -   |    |        |      |        |     |      |     |   +  |       |
+| The metrics exporter `export` function receives a batch of metrics.                                                                                                          |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The metrics exporter `export` function returns `Success` or `Failure`.                                                                                                       |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The metrics exporter provides a `ForceFlush` function.                                                                                                                       |          |    |  -   |    |        |      |        |     |      |     |   +  |       |
```
Comment (jsuereth) [inline spec-compliance-matrix.md:174]: This counts: https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/export/MetricExporter.java#L51

**Review by tigrannajaryan — APPROVED** [review]

**Comment by tigrannajaryan** [comment]
@jsuereth Apologies, I hit the merge button too soon. Want me to revert or fix in a follow up PR?

**Comment by jsuereth** [comment]
@tigrannajaryan I'll do a follow on.

File: spec-compliance-matrix.md (hunk 6 above)
Comment (jkwatson) [reply spec-compliance-matrix.md:169]: that's why I marked it with a `-`. :)

File: spec-compliance-matrix.md (hunk 1 above)
Comment (jkwatson) [reply spec-compliance-matrix.md:113]: well, our instruments have a *type*, but there's no `kind` attribute on them. There is a `kind` available for selecting instruments for views, but I didn't think this is what that was referring to

File: spec-compliance-matrix.md (hunk 7 above)
Comment (jkwatson) [reply spec-compliance-matrix.md:174]: maybe we should rename it to `forceFlush()`? I was hoping we could discuss this stuff before merging. ;)

File: spec-compliance-matrix.md [hunk 8]
```diff
@@ -96,85 +96,85 @@ Disclaimer: this list of features is still a work in progress, please refer to t
...
+| The `View` instrument selection criteria is as specified.                                                                                                                    |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The name of the `View` can be specified.                                                                                                                                     |          |    |  +   |    |        |      |        |     |      |     |   +  |       |
+| The configuration for the metrics stream is as specified.                                                                                                                    |          |    |  ?   |    |        |      |        |     |      |     |   -  |       |
+| The specified logic is used to process `Measurement`s made by an instrument.                                                                                                 |          |    |  ?   |    |        |      |        |     |      |     |   +  |       |
```
Comment (jkwatson) [inline spec-compliance-matrix.md:156]: I didn't have a clue what these were referring to, hence the '?'s

**Pull request merged**